strawberry/permission.py

```diff
@@ -52,16 +64,20 @@ class BasePermission(abc.ABC):
     def has_permission(
         self, source: Any, info: Info, **kwargs: Any
     ) -> Union[bool, Awaitable[bool]]:
+        """Check if the permission should be accepted.
+
+        This method should be overridden by the subclasses.
+        """
         raise NotImplementedError(
             "Permission classes should override has_permission method"
         )
 
+    # TODO: show an example of how to override this method
     def on_unauthorized(self) -> None:
-        """
-        Default error raising for permissions.
-        This can be overridden to customize the behavior.
-        """
+        """Default error raising for permissions.
 
+        This method can be overridden to customize the error raised when the permission is not granted.
+        """
```

Comment:
```suggestion
        This method can be overridden to customize the error raised when the permission is not granted.
        
        Example:
        
        ```python
        from strawberry.permission import BasePermission
        
        
        class CustomPermissionError(PermissionError):
            pass
        
        
        class IsAuthenticated(BasePermission):
            message = "User is not authenticated"
            def has_permission(self, source, info, **kwargs):
                return info.context["user"].is_authenticated
            
            def on_unauthorized(self) -> None:
                raise CustomPermissionError(self.message)
        ```
        """
```